src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -312,6 +324,106 @@ def __call__(self, parser, namespace, values, option_string=None):
     main_command_table[name] = cmd
     main_command_module_map[name] = module_name
 
+def cli_generic_wait_command(module_name, name, getter_op, factory=None, table_transformer=None):
+
+    if not isinstance(getter_op, string_types):
+        raise ValueError("Getter operation must be a string. Got '{}'".format(getter_op))
```

Comment:
Would this be a TypeError? Also, recommend the message contain `type(getter_op)` instead of just getter_op. If someone passed 10 it would say "Got '10'" instead of "Got 'int'".